app/controllers/quick_reservations_controller.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+class QuickReservationsController < ApplicationController
+  before_action :authenticate_user!
+  load_resource :facility, find_by: :url_name
+  load_resource :instrument, through: :facility, find_by: :url_name
+  before_action :get_startable_reservation, only: [:index, :start]
+  before_action :prepare_reservation_data, only: [:new, :create]
+
+  # GET /facilities/:facility_id/instruments/:instrument_id/quick_reservations
+  def index
+    redirect_to(new_facility_instrument_quick_reservation_path) unless @startable || @ongoing
+  end
+
+  def new
+    interval = @instrument.quick_reservation_intervals.first
+    @walkup_available = @instrument.walkup_available?(interval:)
+  end
+
+  def create
+    index = reservation_params[:reservation_index].to_i
+    account = Account.find(reservation_params[:account])
+    reservation = @possible_reservations[index]
+
+    build_order
+
+    # Borrowed from ReservationCreator
+    @order.account = account
+    validator = OrderPurchaseValidator.new(@order_detail)
+    raise OrderPurchaseValidatorError, @order_detail if validator.invalid?
+    reservation.order_detail = @order_detail
+    reservation.save_as_user!(current_user)
+
+    @order_detail.assign_estimated_price(reservation.reserve_end_at)
+    @order_detail.save_as_user!(current_user)
+
+    @order.validate_order!
```

Comment:
There's an `OrderPurchaser` class that `OrdersController` uses to purchase an order. It asks for a bit more than we have here to give. I think these 3 lines are enough to purchase an order, but I could've missed something